projects/fw_102_homework/src/main.c

```diff
@@ -0,0 +1,39 @@
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+
+  ++storage->counter_a;
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a);
+
+  if (storage->counter_a % 2 == 0) {
+    ++storage->counter_b;
+    LOG_DEBUG("Counter B: %i\n", storage->counter_b);
+  }
+
+  soft_timer_start_millis(500, prv_timer_callback, storage, NULL);
```

Comment:
Generally it's a good idea to factor out the `500` constant from here and the `soft_timer_start_millis` call below into a `#define`'d constant with an appropriate name, so we know what it is and so we don't have to change it in multiple places if we ever need to change it. It should be suffixed with the unit name (i.e. `_MS`) so we know what the units are.